pgmpy/estimators/EM.py

```diff
@@ -137,19 +138,26 @@ def _parallel_compute_weights(
     ) -> pd.DataFrame:
         cache: List[pd.DataFrame] = []
         for i in range(offset, min(offset + batch_size, data_unique.shape[0])):
-            v = list(product(*[range(card) for card in latent_card.values()]))
-            latent_combinations = np.array(v, dtype=int)
-            df = data_unique.iloc[[i] * latent_combinations.shape[0]].reset_index(
-                drop=True
-            )
-            for index, latent_var in enumerate(latent_card.keys()):
-                df[latent_var] = latent_combinations[:, index]
+            missing_vars = [
+                var
+                for var in latent_card.keys()
+                if pd.isna(data_unique.iloc[i].get(var))
+            ]
```

Comment:
This is incorrect, `missing_vars` should also consider the variables that have missing values, not just the ones that are specified in `latent_card`, unless you preprocess the `latent_card` to contain those as well.